src/main/java/org/apache/sysml/runtime/matrix/data/SparseBlockCSR.java

```diff
@@ -848,7 +841,54 @@ public String toString() {
 		
 		return sb.toString();
 	}
-	
+
+	@Override
+	public boolean checkValidity(int rlen, int clen, long nnz, boolean strict) {
+		//1. correct meta data
+		if( rlen < 0 || clen < 0 ) {
+			throw new RuntimeException("Invalid block dimensions: "+rlen+" "+clen);
+		}
+
+		//2. correct array lengths
+		if(_size != nnz && _ptr.length < rlen+1 && _values.length < nnz && _indexes.length < nnz ) {
+			throw new RuntimeException("Incorrect array lengths.");
+		}
+
+		//3. non-decreasing row pointers
```

Comment:
please check the tabs, on line `856` and similar places. If this format is the correct one, I will try to adapt.